packages/fuel-indexer/src/executor.rs

```diff
@@ -742,76 +740,21 @@ impl WasmIndexExecutor {
+    ) -> IndexerResult<Self> {
         let uid = manifest.uid();
 
-        match &exec_source {
-            ExecutorSource::Manifest => match manifest.module() {
-                crate::Module::Wasm(ref module) => {
-                    let mut bytes = Vec::<u8>::new();
-                    let mut file = File::open(module).await?;
-                    file.read_to_end(&mut bytes).await?;
-
-                    match WasmIndexExecutor::new(
-                        config,
-                        manifest,
-                        bytes.clone(),
-                        pool,
-                        schema_version,
-                    )
-                    .await
-                    {
-                        Ok(executor) => {
-                            let handle = tokio::spawn(run_executor(
-                                config,
-                                manifest,
-                                executor,
-                                killer.clone(),
-                            ));
-
-                            Ok((handle, ExecutorSource::Registry(bytes), killer))
-                        }
-                        Err(e) => {
-                            error!(
-                                "Could not instantiate WasmIndexExecutor({uid}) from ExecutorSource::Manifest: {e:?}."
-                            );
-                            Err(IndexerError::WasmExecutionInstantiationError)
-                        }
-                    }
-                }
-                crate::Module::Native => {
-                    Err(IndexerError::NativeExecutionInstantiationError)
-                }
-            },
-            ExecutorSource::Registry(bytes) => {
-                match WasmIndexExecutor::new(
-                    config,
-                    manifest,
-                    bytes,
-                    pool,
-                    schema_version,
-                )
-                .await
-                {
-                    Ok(executor) => {
-                        let handle = tokio::spawn(run_executor(
-                            config,
-                            manifest,
-                            executor,
-                            killer.clone(),
-                        ));
-
-                        Ok((handle, exec_source, killer))
-                    }
-                    Err(e) => {
-                        error!("Could not instantiate WasmIndexExecutor({uid}) from ExecutorSource::Registry: {e:?}.");
-                        Err(IndexerError::WasmExecutionInstantiationError)
-                    }
-                }
+        match WasmIndexExecutor::new(config, manifest, wasm_bytes, pool, schema_version)
+            .await
+        {
+            Ok(executor) => Ok(executor),
+            Err(e) => {
+                error!(
+                    "Could not instantiate WasmIndexExecutor({uid}) from ExecutorSource::Manifest: {e:?}."
+                );
```

Comment:
We're no longer instantiating from an `ExecutorSource` here, so this error message should be adjusted.